src/server/input/resync_keyboard_dispatcher.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright © 2021 Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authored by: William Wold <william.wold@canonical.com>
+ */
+
+#ifndef MIR_INPUT_RESYNC_KEYBOARD_DISPATCHER_H_
+#define MIR_INPUT_RESYNC_KEYBOARD_DISPATCHER_H_
+
+#include "mir/input/input_dispatcher.h"
+
+#include <vector>
+
+namespace mir
+{
+namespace input
+{
+class Seat;
+
+class ResyncKeyboardDispatcher : public InputDispatcher
```

Comment:
Having `ResyncKeyboardDispatcher` and `KeyRepeatDispatcher` as separate transformations might be seen as adding to the profusion of objects that makes the Mir codebase hard to approach. But having thought about it I think this is right as they can be tested (or even used) independently.